config_ssm.go

```diff
@@ -0,0 +1,111 @@
+package evergreen
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/pkg/errors"
+)
+
+var (
+	awsConfig                    *aws.Config
+	ssmParameterPath, ssmEnabled = os.LookupEnv("SSM_PARAMETER_PATH")
+	ssmDisabledErr               = errors.New("SSM is not enabled")
+)
+
+func getClient(ctx context.Context) (*ssm.Client, error) {
+	if awsConfig == nil {
+		config, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(DefaultEC2Region),
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "loading AWS config")
+		}
+
+		awsConfig = &config
+	}
+
+	return ssm.NewFromConfig(*awsConfig), nil
+}
+
+func getAllParameters(ctx context.Context) (map[string]string, error) {
+	if !ssmEnabled {
+		return nil, ssmDisabledErr
+	}
+
+	client, err := getClient(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting SSM client")
+	}
+
+	res, err := client.GetParametersByPath(ctx, &ssm.GetParametersByPathInput{
+		Path:           aws.String(ssmParameterPath),
+		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "getting parameters")
+	}
+
+	params := make(map[string]string)
+	for _, param := range res.Parameters {
+		if param.Name == nil && param.Value == nil {
+			continue
+		}
+		sectionID, _ := strings.CutPrefix(*param.Name, ssmParameterPath)
+		params[sectionID] = *param.Value
+	}
+	return params, nil
+}
+
+func decodeParameter(ctx context.Context, target ConfigSection) error {
+	if !ssmEnabled {
+		return ssmDisabledErr
+	}
+
+	client, err := getClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting SSM client")
+	}
+
+	res, err := client.GetParameter(ctx, &ssm.GetParameterInput{
+		Name:           aws.String(fmt.Sprintf("%s%s", ssmParameterPath, target.SectionId())),
```

Comment:
Also, can we namespace the admin sections with a collection name as the prefix (admin)? We'll need separate namespaces for separate usages (like admin settings vs project vars) when we start using Parameter Store more widely, so it would be good to set the convention now rather than fix it later.

Reply:
done. 
There are now two prefixes, one overall (so like if prod and staging are sharing the same AWS account they'll have different prefixes for their settings) and one for section (rn just the `admin` prefix).